internal/runtime/crio/crio.go

```diff
@@ -0,0 +1,178 @@
+	if defaultRuntimeName, ok := config.GetPath([]string{"crio", "runtime", "default_runtime"}).(string); ok && defaultRuntimeName != "" {
+		runtimeNamesForConfig = append(runtimeNamesForConfig, defaultRuntimeName)
+	}
+	for _, r := range runtimeNamesForConfig {
+		if options, ok := config.GetPath([]string{"crio", "runtime", "runtimes", r}).(*toml.Tree); ok {
+			options, _ = toml.Load(options.String())
+			config.SetPath([]string{"crio", "runtime", "runtimes", runtimeName}, options)
+			break
+		}
+	}
+
+	config.SetPath([]string{"crio", "runtime", "runtimes", runtimeName, "runtime_path"}, path)
+	config.SetPath([]string{"crio", "runtime", "runtimes", runtimeName, "runtime_type"}, "vm")
+	config.SetPath([]string{"crio", "runtime", "runtimes", runtimeName, "privileged_without_host_devices"}, "true")
+
+	if setAsDefault {
+		config.SetPath([]string{"crio", "runtime", "default_runtime"}, runtimeName)
+	}
+
+	*c.Tree = config
+	return nil
+}
+
+// DefaultRuntime returns the default runtime for the crio config
+func (c *Config) DefaultRuntime() string {
+	if c == nil || c.Tree == nil {
+		return ""
+	}
+	if runtime, ok := c.GetPath([]string{"crio", "runtime", "default_runtime"}).(string); ok {
+		return runtime
+	}
+	return ""
+}
+
+// RemoveRuntime removes a runtime from the crio config
+func (c *Config) RemoveRuntime(name string) error {
+	if c == nil {
+		return nil
+	}
+
+	config := *c.Tree
+	if runtime, ok := config.GetPath([]string{"crio", "runtime", "default_runtime"}).(string); ok {
+		if runtime == name {
+			err := config.DeletePath([]string{"crio", "runtime", "default_runtime"})
+			if err != nil {
+				return err
+			}
+
+		}
+	}
+
+	runtimeClassPath := []string{"crio", "runtime", "runtimes", name}
+	err := config.DeletePath(runtimeClassPath)
+	if err != nil {
+		return err
+	}
+	for i := 0; i < len(runtimeClassPath); i++ {
+		remainingPath := runtimeClassPath[:len(runtimeClassPath)-i]
+		if entry, ok := config.GetPath(remainingPath).(*toml.Tree); ok {
+			if len(entry.Keys()) != 0 {
+				break
+			}
+			err := config.DeletePath(remainingPath)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	*c.Tree = config
+	return nil
+}
+
+// Save writes the config to the specified path
+func (c *Config) Save() (int64, error) {
+	config := c.Tree
+	output, err := config.Marshal()
+	if err != nil {
+		return 0, fmt.Errorf("unable to convert to TOML: %v", err)
+	}
+	f, err := os.Create(c.Path)
+	if err != nil {
+		return 0, fmt.Errorf("unable to open '%s' for writing: %w", c.Path, err)
+	}
+	defer f.Close()
+
+	n, err := f.Write(output)
+	if err != nil {
+		return 0, fmt.Errorf("unable to write output: %w", err)
+	}
+
+	return int64(n), err
+}
+
+func (c *Config) Restart() error {
+	var args []string
+	args = append(args, "chroot", "/host", "systemctl", "restart", "crio")
+
+	klog.Infof("Restarting crio using systemd")
+
+	//nolint:gosec // TODO: Can we harden this so that there is less risk of command injection
```

Comment:
Instead of adding a `//nolint`, we can allowlist the lint check by adding it to the golangci.yml file.

[Here](https://github.com/NVIDIA/gpu-operator/blob/fcb3424a653032c5844142cc33b4d6728c924023/.golangci.yml#L21) is an example

Reply:
Fixed